### PR TITLE
[Enhancement]Persist run mode and check after restart

### PR DESCRIPTION
**Pull request by abc982627271**
- Save run_mode in version file
- Compare the run_mode in version file with the config value on start
- Compare the run_mode in version file with the helper node on first startup

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: fe/fe-core/src/test/java/com/starrocks/persist/StorageTest.java [hunk 1]
```diff
@@ -113,7 +113,7 @@ public void deleteDir() {
 
     @Test
     public void testConstruct() {
-        Storage storage1 = new Storage(1, "token", "test");
+        Storage storage1 = new Storage(1, "token", "shared_nothing", "test");
```
Comment (kevincai) [inline fe/fe-core/src/test/java/com/starrocks/persist/StorageTest.java:116]: add following check?
```java
Assert.assertEquals("shared_nothing", storage1.getRunMode());
```

**Review by kevincai — DISMISSED** [review]

**Comment by abc982627271** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/persist/Storage.java [hunk 2]
```diff
@@ -67,6 +67,7 @@ public class Storage {
     // version file props keys
     private static final String VERSION_PROP_CLUSTER_ID = "clusterId";
     private static final String VERSION_PROP_TOKEN = "token";
+    private static final String VERSION_PROP_RUN_MODE = "run_mode";
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/persist/Storage.java:70]:
```suggestion
    private static final String VERSION_PROP_RUN_MODE = "runMode";
```

File: fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java [hunk 3]
```diff
@@ -355,6 +371,14 @@ public void getClusterIdAndRoleOnStartup() throws IOException {
                             System.exit(-1);
                         }
                     }
+                    String remoteRunMode = conn.getHeaderField(MetaBaseAction.RUN_MODE);
+                    if (runMode == null && remoteRunMode != null) {
+                        LOG.info("get run_mode from helper node. run_mode={}.", remoteRunMode);
+                        runMode = remoteRunMode;
+                        isVersionFileChanged = true;
+                        storage.reload();
+                    }
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java:380]: check runMode != null && remoteRunMode != null && runMode != remoteRunMode?

File: fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java [hunk 4]
```diff
@@ -272,10 +276,20 @@ public void getClusterIdAndRoleOnStartup() throws IOException {
...
                     storage.setToken(token);
-                    storage.writeClusterIdAndToken();
+                    isVersionFileChanged = true;
                 } else {
                     token = storage.getToken();
                 }
+
+                if (storage.getRunMode() == null) {
+                    runMode = GlobalStateMgr.getRunMode().name().toLowerCase();
+                    LOG.info("new runMode is {}", runMode);
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java:286]:
```suggestion
                    LOG.info("new runMode={}", runMode);
```

**Review by kevincai — DISMISSED** [review]

**Comment by abc982627271** [comment]
run starrocks_fe_unittest

**Comment by abc982627271** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 21 / 47 (44.68%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|11|35|31.43%|[279, 284, 285, 286, 287, 288, 290, 320, 340, 346, 364, 376, 377, 378, 379, 380, 381, 384, 385, 386, 387, 388, 422, 424]|
|:large_blue_circle:|com/starrocks/persist/Storage.java|9|11|81.82%|[175, 176]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|1|100.00%|[]|

**Review by wyb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 21 / 48 (43.75%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|11|35|31.43%|[279, 284, 285, 286, 287, 288, 290, 320, 340, 346, 364, 376, 377, 378, 379, 380, 381, 384, 385, 386, 387, 388, 422, 424]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|2|50.00%|[576]|
|:large_blue_circle:|com/starrocks/persist/Storage.java|9|11|81.82%|[175, 176]|

**Review by wyb — DISMISSED** [review]

**Comment by tracymacding** [comment]
run starrocks_admit_test

**Comment by tracymacding** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 20 / 53 (37.74%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|13|44|29.55%|[278, 282, 310, 326, 329, 331, 334, 335, 336, 343, 344, 346, 347, 348, 367, 379, 380, 382, 385, 386, 388, 423, 424, 425, 427, 428, 429, 430, 697, 698, 706]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|2|50.00%|[576]|
|:large_blue_circle:|com/starrocks/persist/Storage.java|6|7|85.71%|[171]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 20 / 54 (37.04%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|13|45|28.89%|[278, 282, 310, 326, 329, 331, 334, 335, 336, 343, 344, 346, 347, 348, 367, 379, 380, 382, 385, 386, 388, 422, 423, 424, 425, 427, 428, 429, 430, 697, 698, 706]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|1|2|50.00%|[576]|
|:large_blue_circle:|com/starrocks/persist/Storage.java|6|7|85.71%|[171]|

**Review by sduzh — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java [hunk 5]
```diff
@@ -381,13 +413,37 @@ public void getClusterIdAndRoleOnStartup() throws IOException {
...
 
+
+        if (Strings.isNullOrEmpty(runMode)) {
+            if (isFirstTimeStartUp) {
+                runMode = RunMode.getCurrentRunMode().getName();
+                storage.setRunMode(runMode);
+                isVersionFileChanged = true;
+            } else if (RunMode.getCurrentRunMode().isAllowCreateLakeTable()) {
+                LOG.error("Upgrading from a cluster with version less than 3.0 to a cluster with run mode {} of " +
+                          "version 3.0 or above is disallowed. exit exit", RunMode.getCurrentRunMode().getName());
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java:424]:
```suggestion
                          "version 3.0 or above is disallowed. will exit", RunMode.getCurrentRunMode().getName());
```

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java (hunk 5 above)
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/server/NodeMgr.java:424]:
```suggestion
                          "version 3.0 or above is disallowed. will exit", RunMode.getCurrentRunMode().getName());
```

**Comment by tracymacding** [comment]
run starrocks_admit_test

**Comment by tracymacding** [comment]
run starrocks_admit_test

**Comment by tracymacding** [comment]
run starrocks_admit_test

**Comment by tracymacding** [comment]
run starrocks_admit_test

**Comment by sduzh** [comment]
run starrocks_admit_test

**Comment by sduzh** [comment]
run starrocks_admit_test

**Comment by sduzh** [comment]
run starrocks_admit_test

**Comment by tracymacding** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**